### ISSUE TITLE
gif file converted to jpeg when the send button is pressed

### ISSUE THREAD

**Issue by adbenitez** [deltachat/deltachat-desktop#3781]
when I drop a gift to the composer area, or even manually selecting it using the attachment button and then the "image" option, the gift is staged and I can see it is still an animated gif, but when I click the send button, the gif is converted to jpg in the outgoing message, selecting to attach as "file" sends the original gif without converting it, so I guess Desktop is converting images instead of leaving that to Delta Chat core!?

**Specs:**

```
Delta Chat Desktop	1.44.1 (git: v1.33.0-669-gc878fc787)
Delta Chat Core	v1.136.3
SQLite	3.39.4
electron	28.2.3
node	18.18.2
```

this is the gif file I tried:

![tenor-3454536370](https://github.com/deltachat/deltachat-desktop/assets/24558636/e2a9b9bb-bc46-4942-8fde-b8e2c0e099c9)

**Comment by iequidoo** [comment]
For gifs `Viewtype::Gif` must be used by the app. Otherwise the core will convert the gif to jpeg if it's big. There's no such problem on Android, the gif is sent as is if i use the "Gallery" attachment option

**Comment by iequidoo** [comment]
> no, it does not do such things, maybe it needs to detect gif and then set the view type to gif

Yes, that's what i tried to say :) But maybe we should move this detection to the core somehow? Not sure. CC @r10s

**Comment by Simon-Laux** [comment]
maybe it makes sense to detect gif in core and send it as gif view type instead. or we could do it based on file ending/mimetype in desktop, I think this is possible/reasonable to do before the composer refactoring.

**Comment by r10s** [comment]
> For gifs Viewtype::Gif must be used by the app. Otherwise the core will convert the gif to jpeg if it's too big or contains Exif

GIF cannot contain EXIF afaik.

so, maybe better try to send them always as is, independent of their size. in contrast to JPG and PNG they are not created by the user during normal usage of their devices, so if they're large, then it's that - similar to video or music files or stickers [^1] 

iirc, the gist was always to let ui set `Viewtype::Image` or `Viewtype::File` that is then converted to `Viewtype::Gif` by core as needed.

see https://github.com/deltachat/deltachat-core-rust/blob/8df3b1bb1b644dca823b1ad1dd60cbb5a4eda3b7/src/chat.rs#L2536 - guess_msgtype_from_suffix() converts to GIF - but the recent png-to-jpg conversions (which make totally sense and is a huge improvement, to be clear), are, maybe too greedy and converted GIF as well :)

> There's no such problem on Android, a gif is sent as is if i use the "Gallery" attachment option

i was also wondering why android/ios do not have this but - and indeed, they check for the extension/mime-type and set GIF explicitly. still wondering how this is different from core setting "better type" using guess_msgtype_from_suffix(), this needs some research :)

i think, core should also add some explicit tests. we could then maybe also simplify android/ios code

tl;dr: i think, it should be fixed in core 

[^1]: there is also the question if `Viewtype::Gif` makes sense at all and if it cannot be just a sticker (probably yes :), however, for the issue at hand, i would not go for changes 



i would also expect core not to convert GIF to JPEG
